pkg/controllers/provisioning/provisioner.go

```diff
@@ -116,18 +116,14 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	if len(machines) == 0 {
 		return reconcile.Result{}, nil
 	}
-	machineNames, err := p.LaunchMachines(ctx, machines, RecordPodNomination)
-
-	// Any successfully created node is going to have the nodeName value filled in the slice
-	successfullyCreatedNodeCount := lo.CountBy(machineNames, func(name string) bool { return name != "" })
-	metrics.NodesCreatedCounter.WithLabelValues(metrics.ProvisioningReason).Add(float64(successfullyCreatedNodeCount))
+	_, err = p.LaunchMachines(ctx, machines, metrics.ProvisioningReason, RecordPodNomination)
 
 	return reconcile.Result{}, err
 }
 
 // LaunchMachines launches nodes passed into the function in parallel. It returns a slice of the successfully created node
 // names as well as a multierr of any errors that occurred while launching nodes
-func (p *Provisioner) LaunchMachines(ctx context.Context, machines []*scheduler.Machine, opts ...functional.Option[LaunchOptions]) ([]string, error) {
+func (p *Provisioner) LaunchMachines(ctx context.Context, machines []*scheduler.Machine, launchReason string, opts ...functional.Option[LaunchOptions]) ([]string, error) {
```

Comment:
What do you think about making the `launchReason` part of the `LaunchOptions` so you could have a more expressive call to LaunchMachines like

```go
nodeNames, err := c.provisioner.LaunchMachines(ctx, action.replacementNodes, WithReason(reason), RecordPodNomination)
```

Reply:
I didn't like the additional param but thought that `LaunchOptions` was mostly for optional settings that can affect scheduling (enable / disable some actions or optional configuration) - can move it there if that is ok